gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapper.java

```diff
@@ -0,0 +1,171 @@
+import org.hl7.v3.RCMRMT030101UK04Annotation;
+import org.hl7.v3.RCMRMT030101UK04EhrComposition;
+import org.hl7.v3.RCMRMT030101UK04ObservationStatement;
+import org.hl7.v3.RCMRMT030101UK04PertinentInformation02;
+
+import uk.nhs.adaptors.pss.translator.util.DateFormatUtil;
+
+public class ImmunizationMapper {
+    private static final String META_PROFILE = "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Immunization-1";
+    private static final String SNOMED_CODE = "2.16.840.1.113883.2.1.3.2.3.15";
+    private static final String IDENTIFIER_SYSTEM = "https://PSSAdaptor/";
+    private static final String VACCINE_PROCEDURE_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-VaccinationProcedure-1";
+    private static final String END_DATE = "End Date: ";
+    private static final String URL = "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-coding-sctdescid";
+
+    private final CodeableConceptMapper codeableConceptMapper = new CodeableConceptMapper();
+
+
+    public List<Immunization> mapToImmunization(RCMRMT030101UK04ObservationStatement observationStatement, Patient patient,
+        Encounter encounter, Practitioner practitioner, RCMRMT030101UK04EhrComposition ehrComposition) {
+
+        if (checkSnomedCode(observationStatement)) {
+            var id = observationStatement.getId().getRoot();
+            var identifier = getIdentifier(id);
+            var note = buildNotes(Optional.of(observationStatement));
+
+            if (highValueToNotes(observationStatement) != null) {
+                var newNote = new Annotation(new StringType(note.get(0).getText()
+                    + StringUtils.SPACE + (new StringType(END_DATE + highValueToNotes(observationStatement)))));
+                note.add(newNote);
+            }
+
+            var date = getDate(observationStatement);
+
+            String recordedTime;
+
+            if (ehrComposition.getAuthor() != null) {
+                recordedTime = ehrComposition.getAuthor().getTime().getValue();
+            } else if (ehrComposition.getEffectiveTime() != null) {
+                recordedTime = DateFormatUtil.parse(ehrComposition.getAvailabilityTime().getValue()).asStringValue();
+            } else {
+                recordedTime = null;
+            }
+
+            Extension recordedTimeExtension = new Extension()
+                .setValue(new StringType(recordedTime));
+
+            Extension vaccineExtension = createExtension(observationStatement);
+            return createImmunization(id, identifier, note, vaccineExtension, date, recordedTimeExtension);
+        }
+
+        return null;
+    }
+
+    private Identifier getIdentifier(String id) {
+        return new Identifier()
+            .setSystem(IDENTIFIER_SYSTEM) // TODO: concatenate source practice org id to URL (NIAD-2021)
+            .setValue(id);
+    }
+
+    private boolean checkSnomedCode(RCMRMT030101UK04ObservationStatement observationStatement) {
```

Comment:
I think it's a good idea to prefix methods that return booleans with 'is'. This will automatically force you to name it in a more specific way.
'check' is fine for checking conditions and throwing exceptions if mot matched.